src/ui/controls/edit.cpp

```diff
@@ -47,19 +47,20 @@
 namespace Ui
 {
 
-const float MARGX           = (5.0f/640.0f);
-const float MARGY           = (5.0f/480.0f);
-const float MARGYS          = (4.0f/480.0f);
-const float MARGY1          = (1.0f/480.0f);
+const float MARGX            = (5.0f/640.0f);
+const float MARGY            = (5.0f/480.0f);
+const float MARGYS           = (4.0f/480.0f);
+const float MARGY1           = (1.0f/480.0f);
 //! time limit for double-click
-const float DELAY_DBCLICK   = 0.75f;
+const float DELAY_DBCLICK    = 0.75f;
 //! time limit for scroll
-const float DELAY_SCROLL    = 0.1f;
+const float DELAY_SCROLL     = 0.1f;
 //! expansion for \b;
-const float BIG_FONT        = 1.6f;
-
-
-
+const float BIG_FONT         = 1.6f;
+//! start characters
+const int DEFAULT_CHARACTERS = 100;
+//! infinite edit flag
+const int INFINITE_EDIT = -1;

```

Comment:
You don't really need a separate flag, just use `std::numeric_limits<int>::max()`